### PR TITLE
Ninja response files

### PR DESCRIPTION
**Pull request by ayuckhulk**
This PR adds support for `@response` files for llbuild ninja.

Changes:
* `rspfile` and `rspfile_content` rule attributes support
* `${in_newline}` parameter support
* simple test for response file deletion when a rule succeeded and response file content

### CODE REVIEW

File: lib/Commands/NinjaBuildCommand.cpp [hunk 1]
```diff
@@ -1315,6 +1315,27 @@ buildCommand(BuildContext& context, ninja::Command* command) {
 #endif
       }
 
+      // If response file is used by the command, create the file and
+      // fill it with content before command execution.
+      // The file should be deleted after successful command execution.
+      const auto rspFile = command->getRspFile();
+      if (!rspFile.empty())
+      {
```
Comment (ddunbar) [inline lib/Commands/NinjaBuildCommand.cpp:1323]: Nit: `if (!...) {` per LLVM style

File: lib/Ninja/ManifestLoader.cpp [hunk 2]
```diff
@@ -395,11 +395,12 @@ class ManifestLoaderImpl: public ParseActions {
...
       
-    // Support "in" and "out".
-    if (name == "in") {
+    // Support "in", "in_newline" and "out".
+    if (name == "in" || name == "in_newline") {
+      const auto separator = name == "in" ? ' ' : '\n';
       for (unsigned i = 0, ie = decl->getNumExplicitInputs(); i != ie; ++i) {
         if (i != 0)
-          result << " ";
+          result << separator;
```
Comment (ddunbar) [inline lib/Ninja/ManifestLoader.cpp:403]: I have a mild preference for just inlining the `separator` variable, so reader doesn't have to ever look an see if it is used elsewhere, but not a big deal.

File: lib/Ninja/ManifestLoader.cpp [hunk 3]
```diff
@@ -521,7 +522,18 @@ class ManifestLoaderImpl: public ParseActions {
     lookupNamedBuildParameter(decl, startTok, "restat", restat);
     decl->setRestatFlag(!restat.str().empty());
 
-    // FIXME: Handle rspfile attributes.
```
Comment (ddunbar) [inline lib/Ninja/ManifestLoader.cpp:524]: My hero! :)

**Comment by ddunbar** [comment]
@swift-ci please smoke test

File: lib/Ninja/ManifestLoader.cpp (hunk 2 above)
Comment (ayuckhulk) [reply lib/Ninja/ManifestLoader.cpp:403]: Are you suggesting something like this?
```suggestion
          result << (name == "in" ? ' ' : '\n');
```

It seems less readable to me :)

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Comment by dmbryson** [comment]
@swift-ci please smoke test

**Pull request merged**